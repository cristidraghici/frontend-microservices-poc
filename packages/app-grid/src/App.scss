.AppGrid {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  background-color: #f4f4f4;
  color: #333;
  margin: 0;

  table {
    width: 100%;
    border-collapse: collapse;

    th,
    td {
      padding: 0.8rem;
      text-align: left;
      border: 1px solid #bdc3c7;
    }

    th {
      background-color: #3498db;
      color: #fff;
    }

    tbody tr:nth-child(even) {
      background-color: #ecf0f1;
    }
  }
}
