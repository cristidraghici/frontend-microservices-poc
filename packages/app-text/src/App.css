#root {
  margin: 0;
  padding: 0;
  text-align: justify;

  font-size: 3rem;
}
