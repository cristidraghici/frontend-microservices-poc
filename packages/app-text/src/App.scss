.AppText {
  margin: 0;
  padding: 0;
  text-align: justify;

  font-size: 2rem;
}
