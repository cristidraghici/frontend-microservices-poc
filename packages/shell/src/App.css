body, html {
  margin: 0;
  height: 100%;
}

.wrapper {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  height: 100%;
}

section {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #3498db;
  color: #fff;
  font-size: 2rem;
  height: calc(50vh - 20px); /* Set height to 50% of viewport height */
  margin: 10px;

  overflow: scroll;
}

section:nth-child(even) {
  background-color: #2ecc71;
}