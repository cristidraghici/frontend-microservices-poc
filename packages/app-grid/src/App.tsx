import { FunctionComponent, useState, useEffect } from "react";
import "./App.scss";

type User = {
  id: number;
  name: string;
  username: string;
};

const App: FunctionComponent = () => {
  const [users, setUsers] = useState<User[]>([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await fetch(
          "https://jsonplaceholder.typicode.com/users"
        );
        const data = await response.json();
        console.log(data);
        setUsers(data);
      } catch (error) {
        console.error("Error fetching data:", error);
      }
    };

    fetchData();
  }, []);

  return (
    <div className="AppGrid">
      <table>
        <thead>
          <tr>
            <th>ID</th>
            <th>Name</th>
            <th>Username</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user) => (
            <tr key={user.id}>
              <td>{user.id}</td>
              <td>{user.name}</td>
              <td>{user.username}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default App;
