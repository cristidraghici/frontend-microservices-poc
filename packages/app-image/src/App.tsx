import PuppyJPG from "/puppy.jpg";
import "./App.css";

function App() {
  return <img src={PuppyJPG} />;
}

export default App;
