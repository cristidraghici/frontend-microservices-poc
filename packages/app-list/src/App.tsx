import { useState, useEffect } from "react";
import EventBus from "common-state";
import "./App.css";

import { DEFAULT_LIST } from "./constants";

function App() {
  const [list, setList] = useState<string[]>(DEFAULT_LIST);

  useEffect(() => {
    EventBus.subscribe("list:update", (newList: string[]) => {
      setList(newList);
    });
  }, []);

  return (
    <ul>
      {list.map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </ul>
  );
}

export default App;
