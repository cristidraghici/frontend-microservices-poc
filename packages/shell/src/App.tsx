import { FunctionComponent } from "react";
import MicroFrontendContainer from "./components/MicroFrontendContainer";

import { APP_INFO } from "./config";

import "./App.css";

const App: FunctionComponent = () => {
  return (
    <div className="wrapper">
      {APP_INFO.map((app) => (
        <MicroFrontendContainer
          className="MicroFrontend"
          key={app.id}
          id={app.id}
          url={app.url}
        />
      ))}
    </div>
  );
};

export default App;
