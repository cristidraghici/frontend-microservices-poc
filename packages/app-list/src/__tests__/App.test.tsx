import { render } from "@testing-library/react";

import App from "../App";

describe("App", () => {
  it("should test a snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render", () => {
    render(<App />);
    const element = document.querySelector("ul");
    expect(element).toBeInTheDocument();
  });
});
