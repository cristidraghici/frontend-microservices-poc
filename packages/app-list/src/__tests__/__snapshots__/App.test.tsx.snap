// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should test a snapshot 1`] = `
<DocumentFragment>
  <div
    class="AppList"
  >
    <ul>
      <li>
        Default item 1
      </li>
      <li>
        Default item 2
      </li>
      <li>
        Default item 3
      </li>
      <li>
        Default item 4
      </li>
      <li>
        Default item 5
      </li>
      <li>
        Default item 6
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
